### PR TITLE
feat: [PL-24815]: Added new interface for scm api error handling

### PR DESCRIPTION
**Pull request by bhavya181**
Added a new interface that will be implemented by all repoProviders for all APIs and a new enum factory has been added that returns handler based on API and repoProvider

You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)

### CODE REVIEW

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 7 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java` line 11 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0cvuf5346xwnORKONC:-N0cvuf5346xwnORKOND:bnjeqtj) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java#L11)):*
> ```Java
> import lombok.Getter;
> 
> public enum ScmApis {
> ```

Whats the purpose of this class?
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/ScmApiErrorHandlingHelper.java` line 22 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0cv_X44Jz-MWcsa82E:-N0cv_X5ALdqllhmvi5w:bi07fcf) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/ScmApiErrorHandlingHelper.java#L22)):*
> ```Java
> @Slf4j
> public class ScmApiErrorHandlingHelper {
>   @Inject Map<Pair<ScmApis, RepoProviders>, ScmApiErrorHandler> scmApiErrorHandlerMap;
> ```

No need to inject. Directly maintain a map over here. [SCM API -> [SCM Provider -> Error Handler Class]].

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/bitbucket/BitbucketListRepoScmApiErrorHandler.java` line 19 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0dUkEOBWhsl4XwrQGE:-N0dUkEP5EG96CECMOaT:bzegimr) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/bitbucket/BitbucketListRepoScmApiErrorHandler.java#L19)):*
> ```Java
>       case 401:
>       case 403:
>         throw new ScmException(errorMessage, null, SCM_UNAUTHORIZED);
> ```

Lets start creating proper exceptions which inherently understand error codes. eg: ScmUnauthorizedException etc.

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 7 file(s) to review. 2 change requests. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java` line 191 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0dmiRVAIL4de9ni7l0:-N0dmiRVAIL4de9ni7l1:bnrggjl) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java#L191)):*
> ```Java
>       for (RepoProviders repoProvider : RepoProviders.values()) {
>         scmApiErrorHandlerMapBinder.addBinding(Pair.of(scmApiErrorHandler, repoProvider))
>             .to(scmApiErrorHandler.getErrorHandler(repoProvider));
> ```

For every scm API we are using the same error handler here. So why do we want a Pair<ScmApis, RepoProviders> ?
We could have simply used MapBinder< RepoProviders, ScmApiErrorHandler>

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 7 file(s) to review. 2 change requests. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java` line 188 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0dn3G81VDV_QyyE0Gr:-N0dn3G81VDV_QyyE0Gs:b-yjnxd2) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java#L188)):*
> ```Java
>         binder(), new TypeLiteral<Pair<ScmApis, RepoProviders>>() {}, new TypeLiteral<ScmApiErrorHandler>() {});
> 
>     for (ScmApis scmApiErrorHandler : ScmApis.values()) {
> ```

this variable should be named scmApi ?

**Review by DeepakPatankar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 7 file(s) to review. 2 change requests. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java` line 191 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0dnEmH-giOrH-Dw-zJ:-N0dnEmH-giOrH-Dw-zK:b-hlsqfs) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java#L191)):*
> ```Java
>       for (RepoProviders repoProvider : RepoProviders.values()) {
>         scmApiErrorHandlerMapBinder.addBinding(Pair.of(scmApiErrorHandler, repoProvider))
>             .to(scmApiErrorHandler.getErrorHandler(repoProvider));
> ```

Looks like a factory scmApiErrorHandler is already there. Then why do we need a mapbinder ?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 7 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java` line 11 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0cvuf5346xwnORKONC:-N0fKL72AOedIgrfMcHU:b-ymoily) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java#L11)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

Whats the purpose of this class?
</blockquote></details>

This class will give handler based on api and repoProvider
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/ScmApiErrorHandlingHelper.java` line 22 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0cv_X44Jz-MWcsa82E:-N0fKRGg7Oe1tXzS4p1Q:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/helper/ScmApiErrorHandlingHelper.java#L22)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

No need to inject. Directly maintain a map over here. [SCM API -> [SCM Provider -> Error Handler Class]].
</blockquote></details>

Done.
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java` line 188 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0dn3G81VDV_QyyE0Gr:-N0fKTKd9oTFxaKj9yFu:b-5dfryo) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java#L188)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

this variable should be named scmApi ?
</blockquote></details>

removed it
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java` line 191 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0dmiRVAIL4de9ni7l0:-N0fKUv0AVCzsjXW1IbH:bje5540) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java#L191)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

For every scm API we are using the same error handler here. So why do we want a Pair<ScmApis, RepoProviders> ?
We could have simply used MapBinder< RepoProviders, ScmApiErrorHandler>
</blockquote></details>

removed binding
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java` line 191 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0dnEmH-giOrH-Dw-zJ:-N0fKWx63zLJWHu1659J:b-dct48) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/GitSyncModule.java#L191)):*
<details><summary><i>Previously, DeepakPatankar (Deepak Patankar) wrote…</i></summary><blockquote>

Looks like a factory scmApiErrorHandler is already there. Then why do we need a mapbinder ?
</blockquote></details>

this is different and handler is different,. but anyways removed it.

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 7 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java` line 11 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0cvuf5346xwnORKONC:-N0iM3jQ3j40uE6CzsWT:b5afsww) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java#L11)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

This class will give handler based on api and repoProvider
</blockquote></details>

@bhavya181. This map should be maintained in ScmApiErrorHandlingHelper class. Map should be Map<SCMApi, Map<SCMType, ScmApiErrorHandler>> or Map<SCMType, Map<SCMApi, ScmApiErrorHandler>>

This class should just list the APIs. Thats it.

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 7 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java` line 11 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0cvuf5346xwnORKONC:-N0im-I-BDdMTPXGBE5-:b3451lv) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java#L11)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

@bhavya181. This map should be maintained in ScmApiErrorHandlingHelper class. Map should be Map<SCMApi, Map<SCMType, ScmApiErrorHandler>> or Map<SCMType, Map<SCMApi, ScmApiErrorHandler>>

This class should just list the APIs. Thats it.
</blockquote></details>

So, the map will be very large, because for each repoPovider and each API, we have a different handler, so, maintaining a map in ScmApiErrorHandlingHelper will poplute that class, tthat's why created this separate class that will map the things

**Review by ankit-harness — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 8 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java` line 11 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0cvuf5346xwnORKONC:-N0jDxPqDJ-jM4ce20A4:bqgw2s1) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java#L11)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

So, the map will be very large, because for each repoPovider and each API, we have a different handler, so, maintaining a map in ScmApiErrorHandlingHelper will poplute that class, tthat's why created this separate class that will map the things
</blockquote></details>

It's fine Bhavya. Right now, map will have 2-3 entries. Once we see the need of a separate class, we can move the map over there. I want to avoid extra indirection unless required.

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 8 file(s) to review. 2 change requests. 5. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java` line 11 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0cvuf5346xwnORKONC:-N0jENMf2qbDm7szGt7R:b-of3pqn) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/beans/ScmApis.java#L11)):*
<details><summary><i>Previously, ankit-harness (Ankit Singhal) wrote…</i></summary><blockquote>

It's fine Bhavya. Right now, map will have 2-3 entries. Once we see the need of a separate class, we can move the map over there. I want to avoid extra indirection unless required.
</blockquote></details>

There ais only API integrated right now, but there 2-3 more apis that need to be implemented once this framework is ready, like listBranches, getFile and crud for pipeline, so imo framework should be created in right way.

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 14 file(s) to review. 2 change requests. 6. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/bitbucket/BitbucketListRepoScmApiErrorHandler.java` line 19 at r1](https://reviewable.io/reviews/harness/harness-core/32537#-N0dUkEOBWhsl4XwrQGE:-N0klnrg7OmoE0ME4fHS:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5aa85fc3350a6abd02677d6708ff61f66781cdc6/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/bitbucket/BitbucketListRepoScmApiErrorHandler.java#L19)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

Lets start creating proper exceptions which inherently understand error codes. eg: ScmUnauthorizedException etc.
</blockquote></details>

Done.

File: 136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlingHelper.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import com.google.common.annotations.VisibleForTesting;
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+
+@UtilityClass
+@Slf4j
+@OwnedBy(PL)
+public class ScmApiErrorHandlingHelper {
+  public void processAndThrowError(ScmApis scmAPI, ConnectorType connectorType, int statusCode, String errorMessage) {
+    if (statusCode < 400) {
```
Comment (abhinav-harness) [inline 136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlingHelper.java:22]: <300?

**Review by ankit-harness — COMMENTED** [review]
> Reviewed 1 of 13 files at r4.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 13 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 1 approvals obtained.

**Review by ankit-harness — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537#-:-N0nbXdQ7cTvY47HKegl:bnfp4nl)* status: 13 file(s) to review. 1 change requests. 4. unresolved discussions.

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlerFactory.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/32537#-N0nuht33s0x1yhyTdIE:-N0nuht33s0x1yhyTdIF:bkc3em) ([raw file](https://github.com/harness/harness-core/blob/748df87b6bfb8c8cb1f7c0f6615bd7acbf7f69a7/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlerFactory.java#L33)):*
> ```Java
>   public ScmApiErrorHandler getHandler(ScmApis scmApi, RepoProviders repoProvider) {
>     try {
>       return scmApiErrorHandlerMap.get(Pair.of(scmApi, repoProvider)).newInstance();
> ```

It generates new instance everytime on calling getHandler()? Or only first time and then re-injects the same instance once an instance is created?



_Code quote:_
```Java
newInstance();
```

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 14 file(s) to review. 1 change requests. 6. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlerFactory.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/32537#-N0nuht33s0x1yhyTdIE:-N0oBJfs9LoFbFZmXJim:b6epm8) ([raw file](https://github.com/harness/harness-core/blob/748df87b6bfb8c8cb1f7c0f6615bd7acbf7f69a7/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlerFactory.java#L33)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

It generates new instance everytime on calling getHandler()? Or only first time and then re-injects the same instance once an instance is created?
</blockquote></details>

It will create a new object whenever we call getHandler() method
___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlingHelper.java` line 22 at r4](https://reviewable.io/reviews/harness/harness-core/32537#-N0kkxdYz5OJW06cMQig-r4-22:-N0oBPme96abjwc_0w0Q:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/748df87b6bfb8c8cb1f7c0f6615bd7acbf7f69a7/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlingHelper.java#L22)):*
<details><summary><i>Previously, abhinav-harness (Abhinav Singh) wrote…</i></summary><blockquote>

<300?
</blockquote></details>

Done.

**Review by mohitg0795 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 14 file(s) to review. 2 change requests. 6. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlerFactory.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/32537#-N0nuht33s0x1yhyTdIE:-N0oUaIiAOcDIsG2RlCG:b-gji85c) ([raw file](https://github.com/harness/harness-core/blob/748df87b6bfb8c8cb1f7c0f6615bd7acbf7f69a7/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlerFactory.java#L33)):*
<details><summary><i>Previously, bhavya181 (Bhavya Agrawal) wrote…</i></summary><blockquote>

It will create a new object whenever we call getHandler() method
</blockquote></details>

So, then we can't do this right. We can't be creating so many objects just like that for a utility class that can always be re-used. Did @ankit-harness mentioned to use it to you?

**Review by bhavya181 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 14 file(s) to review. 2 change requests. 6. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlerFactory.java` line 33 at r4](https://reviewable.io/reviews/harness/harness-core/32537#-N0nuht33s0x1yhyTdIE:-N0oVJ6C-sJbINXXkuXz:bunsypb) ([raw file](https://github.com/harness/harness-core/blob/748df87b6bfb8c8cb1f7c0f6615bd7acbf7f69a7/136-git-sync-manager/src/main/java/io/harness/gitsync/common/scmerrorhandling/ScmApiErrorHandlerFactory.java#L33)):*
<details><summary><i>Previously, mohitg0795 (Mohit Garg) wrote…</i></summary><blockquote>

So, then we can't do this right. We can't be creating so many objects just like that for a utility class that can always be re-used. Did @ankit-harness mentioned to use it to you?
</blockquote></details>

yes @ankit-harness only told me to create instances to convert class into object.

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 4 files at r2, 11 of 13 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: 1 change requests. 3. unresolved discussions.

**Review by DeepakPatankar — APPROVED** [review]
> Reviewed 1 of 4 files at r2, 11 of 13 files at r4, 1 of 1 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/32537)* status: :shipit: complete! ✓

**Pull request merged**